nilearn/decomposition/canica.py

```diff
@@ -127,13 +127,18 @@ def __init__(self, mask=None, n_components=20, smoothing_fwhm=6,
             mask_strategy=mask_strategy, mask_args=mask_args,
             memory=memory, memory_level=memory_level,
             n_jobs=n_jobs, verbose=verbose)
+
+        if(threshold > n_components):
```

Comment:
From the doc, the given threshold can be None, 'auto' or float. Here, you should also check that the given threshold should be a ```numbers.Number``` to raise an error. Also, in Python I think we don't need parenthesis.